### PR TITLE
feat(epic-b): enhance CommentValidator with verification + deduplication (#4063, #4064)

### PR DESCRIPTION
**Pull request by RC918**
## Summary

This PR enhances the CommentValidator module with two improvements to reduce false positives and noise in MorningAI Reviewer Agent output:

**Issue #4063 - Enhanced Verification Step**: Extended false positive detection to catch more types of incorrect LLM claims:
- Undefined variable claims (e.g., "variable `x` is not defined" when `x` is actually defined in the diff)
- Missing import claims (e.g., "need to import `foo`" when `foo` is already imported)
- Original missing argument detection preserved

**Issue #4064 - Comment Deduplication**: Added deduplication to reduce noise:
- Exact location duplicates (same file + same line)
- Semantic duplicates (>80% message similarity using `SequenceMatcher`)
- New `validate_and_deduplicate_comments()` combined pipeline function

Blueprint Alignment: Section 4.1 "Safe by Design", NOISE BUDGET principle.

## Review & Testing Checklist for Human

- [ ] **Verify regex patterns**: The new `UNDEFINED_VAR_CLAIM_PATTERNS`, `MISSING_IMPORT_CLAIM_PATTERNS`, `IMPORT_PATTERN`, and `VARIABLE_DEF_PATTERN` regexes may have edge cases. Test with real review comments to ensure they don't over-match or under-match.
- [ ] **Integration not included**: The new `validate_and_deduplicate_comments()` function is exported but not yet integrated into `langgraph_orchestrator.py`. Decide if this should be integrated now or in a follow-up PR.
- [ ] **80% similarity threshold**: The `DEDUP_SIMILARITY_THRESHOLD = 0.80` is a reasonable default but may need tuning based on production data.
- [ ] **No unit tests**: Consider adding tests for the new regex patterns and deduplication logic before merging.

**Recommended test plan**: 
1. Run the existing test suite to ensure no regressions
2. Manually test with sample review comments containing "undefined variable" and "missing import" claims against diffs that actually contain those definitions/imports

### Notes
- This PR provides the building blocks for #4063 and #4064 but full integration into the review pipeline may require a follow-up change
- Requested by @RC918
- Link to Devin run: https://app.devin.ai/sessions/2b8d0abec43d403e8f90bd0e4dccaba4

---

> [!NOTE]
> Introduces stronger false-positive filtering and noise reduction for LLM review comments.
> 
> - Extends `CommentValidator` to detect additional false positives: undefined variable and missing import claims; preserves missing-argument checks
> - Adds regex extractors for imports/variable defs and file-scoped diff parsing; returns structured `ValidationStats`
> - Implements comment deduplication (`deduplicate_comments`) with similarity-based grouping and `DeduplicationStats`
> - Provides combined pipeline `validate_and_deduplicate_comments()` and logs pipeline metrics
> - Updates `__init__.py` exports to include new deduplication APIs
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit f12d4b31542faa94389b4a66deca5dd19f340097. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: handoff/20250928/40_App/orchestrator/review_context/comment_validator.py [hunk 1]
```diff
@@ -355,3 +524,220 @@ def validate_review_comments(
...
+        location_key = (file_path, line_number)
+        if file_path and line_number and location_key in seen_locations:
+            dup_comment = comment.copy()
+            dup_comment["_dedup_reason"] = "exact_location_duplicate"
+            duplicates.append(dup_comment)
+            continue
+
+        # 2. Check for semantic duplicate (similar message content)
+        is_semantic_dup = False
+        for existing in unique:
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/review_context/comment_validator.py]: IMPACT: This loop checks every existing unique comment against the current comment for similarity. With large comment sets, this could become inefficient as it's O(n^2) complexity. Suggested fix: Consider using a more efficient data structure or algorithm for similarity comparison, such as approximate nearest neighbor search for large datasets.

File: handoff/20250928/40_App/orchestrator/review_context/comment_validator.py [hunk 2]
```diff
@@ -355,3 +527,220 @@ def validate_review_comments(
...
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/review_context/comment_validator.py]: IMPACT: The current approach for semantic duplicate detection has O(n^2) complexity, which could be inefficient for large comment lists. SUGGESTED FIX: Consider using a more efficient algorithm or data structure for similarity comparison, such as approximate nearest neighbors or hashing techniques.

File: handoff/20250928/40_App/orchestrator/review_context/comment_validator.py [hunk 3]
```diff
@@ -355,3 +527,220 @@ def validate_review_comments(
...
+            similarity = _calculate_similarity(message, existing_message)
+            if similarity >= similarity_threshold:
+                dup_comment = comment.copy()
+                dup_comment["_dedup_reason"] = "semantic_duplicate"
+                dup_comment["_similarity"] = similarity
+                duplicates.append(dup_comment)
+                is_semantic_dup = True
+                duplicate_group_count += 1
+                break
+
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/review_context/comment_validator.py]: IMPACT: This implementation may not properly group duplicate comments. The current approach only checks against the 'unique' list, which contains comments that have already been processed. This could lead to multiple duplicates being added to the 'duplicates' list without proper grouping. For example, if three similar comments appear in sequence, only the first would be detected as a duplicate, and the others would be added as new entries.

Comment (RC918) [inline handoff/20250928/40_App/orchestrator/review_context/comment_validator.py]: IMPACT: The current approach has O(n^2) complexity for the semantic duplication check, which could become a performance issue with large numbers of comments. Each new comment is compared against all existing unique comments, leading to inefficient processing.
